lib/ansible/plugins/strategy/__init__.py

```diff
@@ -481,6 +481,8 @@ def parent_handler_match(target_handler, handler_name):
                                 if original_task.action == 'set_fact':
                                     self._variable_manager.set_nonpersistent_facts(target_host, result_item['ansible_facts'].copy())
                                 else:
+                                    if original_task.action == 'setup':
+                                        target_host._gathered_facts = True
```

Comment:
I'm not sure why/who changed this, but originally we used this fact `module_setup=True` to figure out if 'facts ran', this also allows for users to substitute custom fact modules as 'main fact gathering'.